lib/message_store/postgres/get/select_statement.rb

```diff
@@ -22,14 +22,16 @@ def category_stream?
           is_category_stream ||= StreamName.category?(stream_name)
         end
 
-        def self.build(stream_name, position: nil, batch_size: nil)
-          new(stream_name, position, batch_size)
+        def self.build(stream_name, position: nil, batch_size: nil, where_fragment: nil)
+          new(stream_name, position, batch_size, where_fragment)
         end
 
         def sql
           logger.trace(tag: :sql) { "Composing select statement (Stream: #{stream_name}, Category: #{category_stream?}, Types: #{stream_type_list.inspect}, Position: #{position}, Batch Size: #{batch_size})" }
 
-          statement = <<-SQL
+          formatted_where_clause = where_clause.each_line.to_a.join("  ")
```

Comment:
A quick note: I applied some effort towards ensuring the generated SQL statement had its whitespace formatted for maximum readability. The goal of formatting here is to ensure each line in the where clause is indented by two spaces, so that the statement that gets generated looks correct.